### PR TITLE
Blog: Blog post for scaling OpenTelemetry Collectors using Ansible

### PR DESCRIPTION
**Pull request by ishanjainn**
**Preview**: https://deploy-preview-4140--opentelemetry.netlify.app/blog/2024/scaling-opentelemetry-collectors/

### CODE REVIEW

**Review by chalin — COMMENTED** [review]
Other than the step numbering, as noted by @tiffany76, the indentation is wrong:

> image

Any content of a numbered list item should be indented by as many spaces as the number of characters before the first word of the item. (In your current format, that's 3 spaces, but if/when you fix the formatting, that might change.)

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+To follow this guide, ensure you have:
+
+- Linux hosts.
+- SSH access to each of these Linux hosts.
+- Account permissions to install and configure the OpenTelemetry Collector on these hosts.
+
+## Install the Grafana Ansible collection
+
+The [Grafana Agent role](https://github.com/grafana/grafana-ansible-collection/tree/main/roles/grafana_agent) is available in the Grafana Ansible collection as of the 1.1.0 release.
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:25]:
```suggestion
A [OpenTelemetry Collector role](https://github.com/grafana/grafana-ansible-collection/tree/main/roles/opentelemetry_collector) is provided through the [Grafana Ansible collection](https://docs.ansible.com/ansible/latest/collections/grafana/grafana/) as of the 1.1.0 release.
```

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+Here, we outline a strategy for deploying and managing the OpenTelemetry Collector's scalable instances throughout your infrastructure with Ansible, enhancing your overall monitoring strategy and data visualization capabilities in Grafana Cloud.
+
+## Before you begin
+
+To follow this guide, ensure you have:
+
+- Linux hosts.
+- SSH access to each of these Linux hosts.
+- Account permissions to install and configure the OpenTelemetry Collector on these hosts.
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:21]: I guess the end user also needs to have ansible installed?

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+  ```
+  146.190.208.216    # hostname = ubuntu-01
+  146.190.208.190    # hostname = ubuntu-02
+  137.184.155.128    # hostname = centos-01
+  146.190.216.129    # hostname = centos-02
+  198.199.82.174     # hostname = debian-01
+  198.199.77.93      # hostname = debian-02
+  143.198.182.156    # hostname = fedora-01
+  143.244.174.246    # hostname = fedora-02
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md]: I would recommend to replace these IPs with non-public ones, so something like 10.x.x.x, etc.

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+  ```
+
+  > **Note**: If you are copying the above file, remove the comments (#).
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md]:
```suggestion
{{% alert title="Note" %}}

If you are copying the above file, remove the comments (#).


{{% /alert %}}
```

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+  become: true
+
+  vars:
+    grafana_cloud_api_key: <Your Grafana.com API Key>        # Example - eyxxxxxxxx
+    metrics_username: <prometheus-username>                  # Example - 825019
+    logs_username: <loki-username>                           # Example - 411478
+    prometheus_url: <prometheus-push-url>                    # Example - https://prometheus-us-central1.grafana.net/api/prom/push
+    loki_url: <loki-push-url>                                # Example - https://logs-prod-017.grafana.net/loki/api/v1/push
+    tempo_url: <tempo-push-url>                              # Example - tempo-prod-04-prod-us-east-0.grafana.net:443
+    traces_username: <tempo-username>                        # Example - 411478
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:85]:
```suggestion
```

Make this config independent of Grafana Cloud, we want to have this generally applicable.

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+              username: "{{ traces_username }}"
+              password: "{{ grafana_cloud_api_key }}"
+          basicauth/grafana_cloud_prometheus:
+            client_auth:
+              username: "{{ prometheus_url }}"
+              password: "{{ grafana_cloud_api_key }}"
+          basicauth/grafana_cloud_loki:
+            client_auth:
+              username: "{{ logs_username }}"
+              password: "{{ grafana_cloud_api_key }}"
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:105]:
```suggestion
```

Make this config independent of Grafana Cloud, we want to have this generally applicable.

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+                statements:
+                  - set(attributes["deployment.environment"], resource.attributes["deployment.environment"])
+                  - set(attributes["service.version"], resource.attributes["service.version"])
+
+        otel_collector_exporters:
+          otlp/grafana_cloud_traces:
+            # https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlpexporter
+            endpoint: "{{ tempo_url }}"
+            auth:
+              authenticator: basicauth/grafana_cloud_tempo
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:152]: Replace this with jaeger, or use a debug exporter

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+
+          loki/grafana_cloud_logs:
+            # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/lokiexporter
+            endpoint: "{{ loki_url }}"
+            auth:
+              authenticator: basicauth/grafana_cloud_loki
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:158]: replace this with a debug exporter

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+
+          prometheusremotewrite/grafana_cloud_metrics:
+            # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/prometheusremotewriteexporter
+            endpoint: "{{ prometheus_url }}"
+            add_metric_suffixes: false
+            auth:
+              authenticator: basicauth/grafana_cloud_prometheus
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:165]: replace this with prometheus onprem

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+
+
+        otel_collector_service:
+          extensions: [basicauth/grafana_cloud_tempo, basicauth/grafana_cloud_prometheus, basicauth/grafana_cloud_loki]
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:169]:
```suggestion
```

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+              receivers: [otlp, hostmetrics]
+              processors: [resourcedetection, transform/add_resource_attributes_as_metric_attributes, batch]
+              exporters: [prometheusremotewrite/grafana_cloud_metrics]
+            logs:
+              receivers: [otlp]
+              processors: [resourcedetection, batch]
+              exporters: [loki/grafana_cloud_logs]
+```
+
+> **Note:** You'll need to adjust the configuration to match the specific telemetry you intend to collect and where you plan to forward it. This configuration snippet is a basic example designed for traces, logs, and metrics collected using OTLP and forwarded to Grafana Cloud.
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:185]:
```suggestion
{{% alert title="Note" %}}

You'll need to adjust the configuration to match the specific telemetry you intend to collect and where you plan to forward it. This configuration snippet is a basic example designed for traces, logs, and metrics collected using OTLP and forwarded to Jaeger and Prometheus. 

{{% /alert %}}
```

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+```
+
+## Verifying data ingestion into Grafana Cloud
+
+Once you've deployed the OpenTelemetry Collector and configured it to forward data to Grafana Cloud, you can verify the ingestion:
+
+- Log into your Grafana Cloud instance.
+- Navigate to the **Explore** section.
+- Select your Grafana Cloud Prometheus data source from the dropdown menu.
+- Execute a query to confirm the reception of metrics, for example, `{instance="ubuntu-01"}` for a specific host's metrics.
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:204]:
```suggestion
```

see above, no vendor-specific details

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 13]
```diff
@@ -0,0 +1,212 @@
...
+
+## Visualizing metrics and logs in Grafana
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:206]: Make use of OSS Grafana for visualization or jaeger/prometheus standalone

**Review by svrnm — CHANGES_REQUESTED** [review]
Thanks, see my inline comments.

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 14]
```diff
@@ -175,12 +153,11 @@ Deploy the OpenTelemetry Collector across your hosts by executing:
...
+With data successfully ingested into Prometheus, you can use Grafana to create custom dashboards to visualize the metrics received from your OpenTelemetry Collector's. Utilize Grafana's powerful query builder and visualization tools to derive insights from your data effectively.
 
 - Consider creating dashboards that offer a comprehensive overview of your infrastructure's health and performance.
 - Utilize Grafana's alerting features to proactively manage and respond to issues identified through the OpenTelemetry data.
 
-This guide simplifies the deployment of the OpenTelemetry Collector across multiple Linux hosts using Ansible and illustrates how to visualize collected telemetry in Grafana Cloud. Tailor the Ansible roles, OpenTelemetry Collector configurations, and Grafana dashboards to suit your specific monitoring and observability requirements.
+This guide simplifies the deployment of the OpenTelemetry Collector across multiple Linux hosts using Ansible and illustrates how to visualize collected telemetry in Grafana. Tailor the OpenTelemetry Collector Ansible roles configurations, and Grafana dashboards to suit your specific monitoring and observability requirements.
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md]: this still reads like a marketing blob, I have no issues with promoting OSS grafana here, but ideally this is practical, so how to setup grafana swiftly to see some of the data.

File: content/en/blog/2024/scaling-opentelemetry-collectors.md [hunk 15]
```diff
@@ -0,0 +1,152 @@
+---
+title: 
+  Manage OpenTelemetry Collectors at Scale with Ansible
+linkTitle: OTel Collector with Ansible
+date: 2024-03-12
+author: '[Ishan Jain](https://github.com/ishanjainn) (Grafana)'
```
Comment (svrnm) [inline content/en/blog/2024/scaling-opentelemetry-collectors.md:6]: add a cSpell:ignore: section for words unknown to cspell, check other blog posts for examples

File: content/en/blog/2024/scaling-opentelemetry-collectors.md (hunk 14 above)
Comment (svrnm) [reply content/en/blog/2024/scaling-opentelemetry-collectors.md]: yes, exactly.

**Comment by svrnm** [comment]
> Hey @svrnm The REFCACHE workflow tell me to run `npm run test-and-fix` but that doesnt seem to do anything for me

do not worry too much about that, we fix the checks at the end before merging the PR, there are some nuances with links that sometimes break the pipeline, and while updating the content you may break it again and again

> also for the `canonical_url` This can be added later right as I would be publishing a similar blog (for Grafana Cloud usage as OSS is covered here) but on Grafana's Blog

So, here is my semi-professional SEO opinion: the content that comes first is the "canonical url", so the grafana blog should point to this blog as canoncial. If the grafana blog has a policy against that, we need to discuss that.

**Comment by svrnm** [comment]
> > So, here is my semi-professional SEO opinion: the content that comes first is the "canonical url", so the grafana blog should point to this blog as canoncial. If the grafana blog has a policy against that, we need to discuss that.
> 
> Ahh, I don't think that is possible, Ill get that if possible too but would be nice to have it from here aswell, That is possible to update later or needs to be in this PR?

Do you think that it is not possible or do you know that it is not possible? Can you please check with the Grafana Blog Team first and then follow up with us?

Our current policy does not cover this use case, so we would need look into this (cc @open-telemetry/docs-approvers). 

Personally I still think it doesn't feel right to update an existing blog post to point to a blog post that came later and by that downgrading it from a SEO perspective: whoever takes the blog post first has the major work of copyediting and reviewing.

**Pull request closed without merging**